tracetypes/org.eclipse.tracecompass.incubator.ftrace.core/src/org/eclipse/tracecompass/incubator/internal/ftrace/core/trace/BinaryFTrace.java

```diff
@@ -115,6 +115,14 @@ public void initTrace(IResource resource, String path, Class<? extends ITmfEvent
         setFile(file);
     }
 
+    @Override
+    public synchronized void dispose() {
+        super.dispose();
+        // release (indirect) references to mem-mapped file buffers so that tracecompass
+        // can garbage-collect them and unlock the file (e.g. if the user wants to delete it)
+        fStrategy = null;
```

Comment:
There should be a dispose method in the IBinaryFTraceStrategy that does this for the BinaryFTraceV6Strategy.

Reply:
Thanks for the review, done: I moved the clearing of the reference to the strategy class.
I did not go deeper in the structure, let me know if it's good like this.

Reply:
seems I broke something with that, need to look into it :-)

Reply:
The change is perfect, it just needs to pass the test now :)

Reply:
Seems like it needs a null check before calling dispose().

Reply:
Thanks for the hint! Took me a while to find the right logs